include/sys/arc_impl.h

```diff
@@ -176,6 +176,206 @@ typedef struct l1arc_buf_hdr {
 	abd_t			*b_pabd;
 } l1arc_buf_hdr_t;
 
+enum {
+	L2ARC_DEV_HDR_EVICT_FIRST = (1 << 0)	/* mirror of l2ad_first */
+};
+
+/*
+ * Pointer used in persistent L2ARC (for pointing to log blocks & ARC buffers).
+ */
+typedef struct l2arc_log_blkptr {
+	uint64_t	lbp_daddr;	/* device address of log */
```

Comment:
How about something like, `offset within the device, in <units>` (`units` might be bytes, 512-byte units, or ashift).

Reply:
I do not see a clear advantage to this. Could you elaborate?

Reply:
I mean that lbp_daddr is directly compared to l2ad_start and l2ad_end, and set to l2ad_hand. All three l2ad_* variables are already defined in module/zfs/arc.c. Changing the format of lbp_daddr would require defining additional variables for these interactions.

Reply:
I'm confused by the phrase "device address of log".  Is it talking about a `l2arc_dev_t*` (the address of the l2arc device)?  Can we describe it more specifically using wording like, `offset within the device, in <units>`, or `offset in bytes after l2ad_start`?

Reply:
Oh yes, of course. It is offset within the device, in bytes. It is not offset in bytes after l2ad_start. I will update the comment.